v2/rust/src/environment.rs

```diff
@@ -1,5 +1,234 @@
+use super::child_process_supervisor::{ChildProcessOutcome, ChildProcessSupervisor};
+use super::config::{Config, EnvironmentConfig};
+use super::results::{EnvironmentBuildStatesAdministrator, EnvironmentBuildStatus};
+use super::termination::TerminationFlag;
+use anyhow::{Context, Result};
+use log::{debug, error, info};
 use std::path::{Path, PathBuf};
+use std::process::Command;
 
 pub fn environment_building_stdio_directory(working_directory: &Path) -> PathBuf {
     working_directory.join("environment_building_stdio")
 }
+
+pub fn build_environments(config: &Config, termination_flag: &TerminationFlag) -> Result<()> {
+    let suites = config.suites();
+    let mut environment_build_states_administrator =
+        EnvironmentBuildStatesAdministrator::new_with_pending(
+            suites
+                .iter()
+                .map(|(suite_name, _suite_config)| suite_name.to_owned()),
+            &config.working_directory,
+            &config.results_directory,
+        );
+    environment_build_states_administrator.write_atomic()?;
+    let env_building_stdio_directory =
+        environment_building_stdio_directory(&config.working_directory);
+
+    for (suite_name, suite_config) in suites {
+        match Environment::new(suite_name, &suite_config.environment_config).build_instructions() {
+            Some(mut build_instructions) => {
+                info!("Building environment for suite {}", suite_name);
+                environment_build_states_administrator
+                    .insert_and_write_atomic(suite_name, EnvironmentBuildStatus::InProgress)?;
+                configure_stdio_of_environment_build(
+                    &env_building_stdio_directory,
+                    suite_name,
+                    &mut build_instructions.command,
+                )
+                .context("Configuring stdio of environment build process failed")?;
```

Comment:
OTI: Is this really correct? I think failing to build one environment should not affect the remaining tests. Better build everything, otherwise the user has to reconfigure till next error.

Reply:
I am not 100% sure about it either. However, if this happens, there is anyway a severe problem (we were unable to open files). So most likely, it doesn't matter if we crash here or later on, since nothing would run anyway. If this turns out to be a problem, we can change it.